File: Cpp/Calculate_Money_in_Leetcode_Bank.cpp
class Solution {
public:
    int totalMoney(int n) {
        int weeks = n / 7;
        int rem = n % 7;
        int total = 7 * weeks * (weeks + 1) / 2 + 21 * weeks;
        int remaining = rem * (weeks + 1) + rem * (rem - 1) / 2;
        return total + remaining;
    }
};

File: Rust/Calculate_Money_in_Leetcode_Bank.rs
impl Solution {
    pub fn total_money(n: i32) -> i32 {
        let weeks = n / 7;
        let rem = n % 7;
        let total = 7 * weeks * (weeks + 1) / 2 + 21 * weeks;
        let remaining = rem * (weeks + 1) + rem * (rem - 1) / 2;
        total + remaining
    }
}
